packages/teams-js/test/public/liveShare.spec.ts

```diff
@@ -0,0 +1,27 @@
+import { LiveShareHost } from '../../src/internal/liveShareHost';
+import { liveShare } from '../../src/public/liveShare';
+
+describe('MicrosoftTeams-LiveShare', () => {
+  describe('initialize', () => {
```

Comment:
Since you have some logic guarding against calling initialize incorrectly (can't call while it's already initializing or after it's finished initializing) it would be good to have unit tests around that.

Reply:
We have an issue right now that we can't even test the positive test case where Live Share initializes. To do that a couple of things would need to happen.

- teams-js would need to take a `devDependency` on live-share but given we can't publish a version of live-share that works with this code until teams-js has been updated, we have a chicken & egg problem. Someone has to publish first...

Reply:
I wrote this elsewhere, but I think using dependency injection or something similar would let you isolate the part of your code that is creating the LiveShareClient so that you could test the rest of the logic.

Reply:
Removed namespace per discussion